commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java

```diff
@@ -444,7 +444,7 @@ public String toStringExcludingParam(String... params) {
     */
    public boolean isBinary() {
       String customType = getClassType();
-      if (customType == null) return !this.match(MediaType.APPLICATION_OBJECT);
+      if (customType == null) return !this.withoutParameters().match(MediaType.APPLICATION_OBJECT);
```

Comment:
@gustavonalle I think this makes sense here, but let me know if I am missing something. Could you also take a quick look at `server/core/backup/resources/CacheResource.java ` to make sure there isn't any obvious issues with the encoding logic used when converting key/values to byte[] when creating the `CacheBackupEntry`.

Reply:
```match()```already ignores parameters, what is the test case that ```isBinary``` returns wrong results?

Reply:
Without this change calls to `isBinary` on MediaType `application/x-java-object;type=java.lang.String` returns true